src/Symfony/Component/Serializer/Encoder/YamlEncoder.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Encoder;
+
+use Symfony\Component\Yaml\Dumper;
+use Symfony\Component\Yaml\Parser;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Encodes YAML data.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class YamlEncoder implements EncoderInterface, DecoderInterface
```

Comment:
Here you do Encode and Decode both process inside this class. So you class name should be change like `YamlDataConverter`